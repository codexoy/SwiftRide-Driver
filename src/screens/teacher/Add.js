import React from "react"
import { View, Text, Button, StyleSheet, ScrollView } from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import Notification from "../../components/Notification"
import PageTitle from "../../components/PageTitle"
import NotificationButtons from "./NotificationButtons"

class Add extends React.Component {
	static navigationOptions = () => {
		return {
			title: "add",
			tabBarLabel: strings("tabs.add"),
			tabBarAccessibilityLabel: strings("tabs.add"),
			tabBarTestID: "AddTab"
		}
	}
	render() {
		return <ScrollView style={styles.container} />
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	notifications: {
		flex: 1,
		paddingRight: 30,
		paddingLeft: 30,
		marginTop: 20
	},
	notification: {}
})

export default connect()(Add)
