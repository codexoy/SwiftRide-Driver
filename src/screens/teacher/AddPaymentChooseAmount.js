import React from "react"
import {
	View,
	StyleSheet,
	Text,
	TouchableHighlight,
	TouchableOpacity,
	TextInput,
	KeyboardAvoidingView,
	Platform
} from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import { Icon } from "react-native-elements"
import PageTitle from "../../components/PageTitle"
import {
	MAIN_PADDING,
	floatButtonOnlyStyle,
	DEFAULT_MESSAGE_TIME,
	fullButton
} from "../../consts"
import { API_ERROR } from "../../reducers/consts"
import { NavigationActions } from "react-navigation"
import SlidingMessage from "../../components/SlidingMessage"
import { fetchOrError } from "../../actions/utils"
import { popLatestError } from "../../actions/utils"

export class AddPaymentChooseAmount extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			amount: "000",
			slidingMessageVisible: false
		}

		this.addPayment = this.addPayment.bind(this)
	}

	componentDidUpdate() {
		const error = this.props.dispatch(popLatestError(API_ERROR))
		if (error) {
			this.setState({
				error,
				slidingMessageVisible: true
			})
		}
	}

	addPayment = async () => {
		const resp = await this.props.dispatch(
			fetchOrError("/teacher/add_payment", {
				method: "POST",
				body: JSON.stringify({
					amount: parseInt(this.state.amount),
					student_id: this.props.navigation.getParam("student")
						.student_id
				})
			})
		)
		if (resp) {
			this.setState({ error: "", slidingMessageVisible: true }, () => {
				setTimeout(
					() =>
						this.props.navigation.dispatch(
							NavigationActions.back()
						),
					DEFAULT_MESSAGE_TIME
				)
			})
		}
	}

	changeAmount = amount => {
		this.setState({ amount })
	}

	render() {
		return (
			<View style={styles.container}>
				<SlidingMessage
					visible={this.state.slidingMessageVisible}
					error={this.state.error}
					success={strings("teacher.add_payment.success")}
					close={() =>
						this.setState({ slidingMessageVisible: false })
					}
				/>
				<View style={styles.headerRow}>
					<TouchableOpacity
						onPress={() => {
							this.props.navigation.goBack()
						}}
					>
						<Icon name="arrow-forward" type="material" />
					</TouchableOpacity>
					<PageTitle
						style={styles.title}
						title={strings("teacher.add_payment.title2")}
					/>
				</View>
				<KeyboardAvoidingView
					behavior={Platform.OS === "ios" ? "position" : "height"}
					keyboardVerticalOffset={Platform.select({
						ios: 62,
						android: 24
					})}
					style={styles.container}
				>
					<View style={styles.amountContainer}>
						<TextInput
							placeholder="000.00"
							value={this.state.amount}
							onChangeText={amount => this.changeAmount(amount)}
							style={styles.amountInput}
							autoFocus={true}
							maxLength={5}
							keyboardType="number-pad"
						/>
					</View>
					<TouchableHighlight
						underlayColor="#ffffff00"
						onPress={this.addPayment}
						style={styles.floatButton}
					>
						<Text style={styles.buttonText}>{strings("done")}</Text>
					</TouchableHighlight>
				</KeyboardAvoidingView>
			</View>
		)
	}
}

function mapStateToProps(state) {
	return {
		fetchService: state.fetchService,
		errors: state.errors
	}
}
export default connect(mapStateToProps)(AddPaymentChooseAmount)

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	title: {
		marginLeft: 8,
		marginTop: -4
	},
	headerRow: {
		flexDirection: "row",
		padding: 0,
		justifyContent: "flex-start",
		marginLeft: MAIN_PADDING,
		marginRight: MAIN_PADDING,
		marginTop: 20
	},
	amountInput: {
		fontSize: 80,
		alignItems: "center",
		fontFamily: "Assistant-Light"
	},
	amountContainer: {
		borderBottomWidth: 0,
		backgroundColor: "#f4f4f4",
		padding: 20,
		marginTop: 80
	},
	buttonText: {
		color: "#fff",
		fontWeight: "bold",
		fontSize: 20
	},
	floatButton: {
		...fullButton
	}
})
