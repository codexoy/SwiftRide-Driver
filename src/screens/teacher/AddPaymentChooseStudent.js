import React from "react"
import {
	View,
	StyleSheet,
	Text,
	FlatList,
	TouchableHighlight
} from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import Row from "../../components/Row"
import UserWithPic from "../../components/UserWithPic"
import Separator from "../../components/Separator"
import { SearchBar, Button, Icon } from "react-native-elements"
import PageTitle from "../../components/PageTitle"
import { MAIN_PADDING } from "../../consts"
import { getStudents } from "../../actions/students"
import { NavigationActions } from "react-navigation"
import EmptyState from "../../components/EmptyState"
import StudentsLoader from "../../components/StudentsLoader"

export class AddPaymentChooseStudent extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			search: "",
			student: {},
			students: [],
			page: 1,
			nextUrl: "",
			loading: true
		}

		this.renderItem = this.renderItem.bind(this)
		this._getStudents()
	}

	updateSearch = search => {
		this.setState({ search, loading: true }, () => {
			this._getStudents(false)
		})
	}

	_getStudents = async (append = true) => {
		resp = await getStudents(this.props.fetchService, this.state)
		let newValue = resp.students
		if (append) {
			newValue = [...this.state.students, ...newValue]
		}
		this.setState({
			students: newValue,
			nextUrl: resp.nextUrl,
			loading: false
		})
	}

	endReached = () => {
		if (!this.state.nextUrl) return
		this.setState(
			{
				page: this.state.page + 1
			},
			() => {
				this._getStudents()
			}
		)
	}

	onPress = item => {
		this.setState(
			{
				student: item
			},
			() => {
				this.props.navigation.navigate("Second", { student: item })
			}
		)
	}

	renderItem = ({ item, index }) => {
		let noneMargin = {}
		if (index == 0) {
			noneMargin = { marginTop: 0 }
		}
		return (
			<TouchableHighlight
				underlayColor="#f9f9f9"
				key={`student${item.student_id}`}
				onPress={() => this.onPress(item)}
			>
				<Row
					leftSide={
						<View style={styles.addButton}>
							<Icon name="ios-add" type="ionicon" color="#000" />
						</View>
					}
					style={{ ...styles.row, ...noneMargin }}
				>
					<UserWithPic
						name={item.user.name}
						nameStyle={styles.nameStyle}
						width={64}
						height={64}
					/>
				</Row>
			</TouchableHighlight>
		)
	}

	_renderEmpty = () => (
		<EmptyState
			image="students"
			text={strings("no_students")}
			style={styles.empty}
		/>
	)

	_renderStudents = () => {
		if (this.state.loading) {
			return (
				<View style={styles.listLoader}>
					<StudentsLoader />
				</View>
			)
		}
		return (
			<FlatList
				data={this.state.students}
				keyboardShouldPersistTaps="always"
				renderItem={this.renderItem}
				keyExtractor={item => `student${item.student_id}`}
				onEndReached={this.endReached}
				keyboardDismissMode="on-drag"
				keyboardShouldPersistTaps="always"
				ListEmptyComponent={this._renderEmpty}
			/>
		)
	}
	render() {
		return (
			<View style={styles.container}>
				<PageTitle
					style={styles.title}
					title={strings("teacher.add_payment.title1")}
					leftSide={
						<Button
							icon={
								<Icon
									name="ios-close"
									type="ionicon"
									size={36}
								/>
							}
							onPress={() => {
								this.props.navigation.dispatch(
									NavigationActions.back()
								)
							}}
							type="clear"
							style={styles.closeButton}
						/>
					}
				/>
				<View
					style={styles.studentsSearchView}
					testID="StudentsSearchView"
				>
					<SearchBar
						placeholder={strings("teacher.students.search")}
						onChangeText={this.updateSearch}
						value={this.state.search}
						platform="ios"
						containerStyle={styles.searchBarContainer}
						inputContainerStyle={styles.inputContainerStyle}
						cancelButtonTitle={strings("teacher.students.cancel")}
						inputStyle={styles.search}
						textAlign="right"
						autoFocus={true}
						testID="searchBar"
					/>
					<Separator />
					{this._renderStudents()}
				</View>
			</View>
		)
	}
}

function mapStateToProps(state) {
	return {
		fetchService: state.fetchService
	}
}

export default connect(mapStateToProps)(AddPaymentChooseStudent)

const styles = StyleSheet.create({
	container: {
		flex: 1,
		marginLeft: MAIN_PADDING,
		marginRight: MAIN_PADDING,
		marginTop: 20
	},
	studentsSearchView: { paddingTop: 0 },
	searchBarContainer: {
		backgroundColor: "transparent",
		paddingBottom: 0,
		paddingTop: 0,
		marginTop: 16
	},
	inputContainerStyle: {
		borderRadius: 30,
		paddingLeft: 8,
		paddingTop: 6,
		paddingBottom: 6,
		width: "100%",
		marginLeft: 0,
		marginRight: 0
	},
	search: {
		alignItems: "flex-start",
		paddingLeft: 6,
		fontSize: 14,
		marginLeft: 0
	},
	nameStyle: {
		fontSize: 18,
		marginTop: 14
	},
	addButton: {
		marginTop: 12
	},
	row: { marginTop: 20 },
	closeButton: {
		marginTop: -6
	}
})
