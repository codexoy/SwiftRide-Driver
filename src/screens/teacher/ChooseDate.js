import React from "react"
import {
	View,
	Text,
	TouchableOpacity,
	StyleSheet,
	FlatList
} from "react-native"
import { connect } from "react-redux"
import { strings, dates } from "../../i18n"
import { CalendarList } from "react-native-calendars"
import ShadowRect from "../../components/ShadowRect"
import Row from "../../components/Row"
import UserWithPic from "../../components/UserWithPic"
import Separator from "../../components/Separator"
import { Icon } from "react-native-elements"
import {
	MAIN_PADDING,
	calendarTheme,
	fullButton,
	SHORT_API_DATE_FORMAT
} from "../../consts"
import Hours from "../../components/Hours"
import { getDateAndString } from "../../actions/lessons"
import EmptyState from "../../components/EmptyState"
import LessonsLoader from "../../components/LessonsLoader"
import moment from "moment"

export class ChooseDate extends React.Component {
	constructor(props) {
		super(props)
		const date = new Date()
		this.state = {
			day: date,
			selected: date.toJSON().slice(0, 10),
			items: [],
			loading: true
		}
		this._getItems(date)
		this.onDayPress = this.onDayPress.bind(this)
	}
	componentDidMount() {
		this.willFocusSubscription = this.props.navigation.addListener(
			"willFocus",
			payload => {
				this._getItems(this.state.day)
			}
		)
	}

	componentWillUnmount() {
		this.willFocusSubscription.remove()
	}

	renderArrow = direction => (
		<Icon
			name={
				direction === "left"
					? "keyboard-arrow-right"
					: "keyboard-arrow-left"
			}
			type="material"
		/>
	)
	renderItem = ({ item, index }) => {
		let student = strings("teacher.no_student_applied")
		if (item.student) {
			student = item.student.name
		}
		return (
			<Row
				key={`item${item.id}`}
				style={styles.lessonRow}
				leftSide={
					<Hours
						style={styles.hours}
						duration={item.duration}
						date={item.date}
					/>
				}
			>
				<UserWithPic
					name={student}
					user={item.student}
					nameStyle={styles.nameStyle}
					width={42}
					height={42}
				/>
			</Row>
		)
	}
	_getItems = async date => {
		const dateObject = getDateAndString(date)
		const resp = await this.props.fetchService.fetch(
			"/appointments/?is_approved=true&date=ge:" +
				dateObject.date.startOf("day").toISOString() +
				"&date=le:" +
				dateObject.date.endOf("day").toISOString(),
			{ method: "GET" }
		)
		this.setState({
			items: resp.json["data"],
			loading: false
		})
	}
	onDayPress = day => {
		this.setState(
			{
				day: day,
				selected: day.dateString,
				loading: true
			},
			() => {
				this._getItems(day)
			}
		)
	}

	_renderEmpty = () => (
		<EmptyState
			image="lessons"
			text={strings("empty_lessons")}
			imageSize="small"
		/>
	)

	_renderLessons = () => {
		if (this.state.loading) {
			return (
				<View style={styles.listLoader}>
					<LessonsLoader width={340} />
				</View>
			)
		}
		return (
			<FlatList
				ItemSeparatorComponent={() => <Separator />}
				ListEmptyComponent={this._renderEmpty}
				testID="scheduleList"
				data={this.state.items}
				renderItem={this.renderItem}
				style={styles.flatList}
				keyExtractor={item => `item${item.id}`}
			/>
		)
	}
	render() {
		return (
			<View style={styles.container}>
				<CalendarList
					current={this.state.selected || new Date()}
					style={styles.calendar}
					theme={calendarTheme}
					// Enable horizontal scrolling, default = false
					horizontal={true}
					// Enable paging on horizontal, default = false
					pagingEnabled={true}
					// Minimum date that can be selected, dates before minDate will be grayed out. Default = undefined
					minDate={new Date().getDate() - 14}
					maxDate={moment()
						.add(4, "months")
						.format(SHORT_API_DATE_FORMAT)}
					markedDates={{
						[this.state.selected]: {
							selected: true,
							disableTouchEvent: true
						}
					}}
					// Handler which gets executed on day press. Default = undefined
					onDayPress={this.onDayPress}
					// Month format in calendar title. Formatting values: http://arshaw.com/xdate/#Formatting
					monthFormat={"MMMM yy"}
					renderArrow={this.renderArrow}
					hideArrows={false}
					// Do not show days of other months in month page.
					hideExtraDays={false}
				/>
				<View style={styles.scheduleContainer}>
					<Text testID="dateString" style={styles.scheduleTitle}>
						{dates("date.formats.short", this.state.selected)}
					</Text>
					<ShadowRect style={styles.schedule}>
						{this._renderLessons()}
					</ShadowRect>
				</View>
				<TouchableOpacity
					onPress={() => {
						this.props.navigation.navigate("Lesson", {
							date: this.state.selected
						})
					}}
					style={fullButton}
					testID="continueButton"
				>
					<Text style={styles.buttonText}>
						{strings("teacher.new_lesson.continue")}
					</Text>
				</TouchableOpacity>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	calendar: {
		flex: 1
	},
	scheduleContainer: {
		flex: 1,
		marginTop: 20
	},
	scheduleTitle: {
		fontSize: 18,
		fontWeight: "bold",
		color: "rgb(121, 121, 121)",
		alignSelf: "flex-start",
		marginLeft: MAIN_PADDING
	},
	schedule: {
		flex: 1,
		marginTop: 24
	},
	hours: {
		marginTop: 4
	},
	nameStyle: {
		fontSize: 18,
		marginTop: 4
	},
	lessonRow: {
		marginTop: 12
	},
	buttonText: {
		color: "#fff",
		fontSize: 20,
		fontWeight: "bold"
	},
	flatList: {
		flex: 1,
		marginBottom: 40,
		alignSelf: "center",
		width: "100%"
	}
})

mapStateToProps = state => {
	return {
		fetchService: state.fetchService
	}
}
export default connect(mapStateToProps)(ChooseDate)
