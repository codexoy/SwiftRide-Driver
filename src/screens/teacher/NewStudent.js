import React from "react"
import {
	View,
	Text,
	TouchableHighlight,
	StyleSheet,
	FlatList
} from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import Row from "../../components/Row"
import UserWithPic from "../../components/UserWithPic"
import Separator from "../../components/Separator"
import { SearchBar, Button, Icon } from "react-native-elements"
import PageTitle from "../../components/PageTitle"

export class NewStudent extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			search: ""
		}
	}
	updateSearch = search => {
		this.setState({ search })
	}

	render() {
		return (
			<View style={styles.container}>
				<View style={styles.headerRow}>
					<Button
						icon={<Icon name="arrow-forward" type="material" />}
						onPress={() => {
							this.props.navigation.goBack()
						}}
						type="clear"
					/>
					<PageTitle
						style={styles.title}
						title={strings("teacher.students.add")}
					/>
				</View>
				<View
					style={styles.studentsSearchView}
					testID="StudentsSearchView"
				>
					<SearchBar
						placeholder={strings("teacher.students.search")}
						onChangeText={this.updateSearch}
						value={this.state.search}
						platform="ios"
						containerStyle={styles.searchBarContainer}
						inputContainerStyle={styles.inputContainerStyle}
						cancelButtonTitle={strings("teacher.students.cancel")}
						inputStyle={styles.search}
						textAlign="right"
						autoFocus={true}
					/>
					<Separator />
					<FlatList
						data={[{ title: "Title Text", key: "item1" }]}
						renderItem={({ item }) => (
							<Row
								style={styles.row}
								leftSide={
									<Icon
										style={styles.arrow}
										name="ios-add"
										type="ionicon"
										color="#000"
									/>
								}
							>
								<UserWithPic
									name="רונן רוזנטל"
									extra={`${strings(
										"teacher.students.lesson_num"
									)}: 13`}
									nameStyle={styles.nameStyle}
									width={54}
									height={54}
									style={styles.userWithPic}
									imageContainerStyle={
										styles.imageContainerStyle
									}
								/>
							</Row>
						)}
					/>
				</View>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	title: {
		marginLeft: 12,
		marginTop: 4
	},
	headerRow: {
		flexDirection: "row",
		flex: 1,
		maxHeight: 50
	},
	studentsSearchView: { padding: 26, paddingTop: 0 },
	searchBarContainer: {
		backgroundColor: "transparent",
		paddingBottom: 0,
		paddingTop: 0
	},
	inputContainerStyle: {
		borderRadius: 30,
		paddingLeft: 8,
		paddingTop: 6,
		paddingBottom: 6,
		width: "100%",
		marginLeft: 0,
		marginRight: 0
	},
	search: {
		alignItems: "flex-start",
		paddingLeft: 6,
		fontSize: 14,
		marginLeft: 0
	}
})

export default connect()(NewStudent)
