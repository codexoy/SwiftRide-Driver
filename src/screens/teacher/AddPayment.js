import { createStackNavigator } from "react-navigation"
import AddPaymentChooseStudent from "./AddPaymentChooseStudent"
import AddPaymentChooseAmount from "./AddPaymentChooseAmount"
export default createStackNavigator(
	{ First: AddPaymentChooseStudent, Second: AddPaymentChooseAmount },
	{
		initialRouteKey: "First",
		headerMode: "none",
		navigationOptions: {
			headerVisible: false
		}
	}
)
