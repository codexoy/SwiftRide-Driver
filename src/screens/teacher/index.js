import React from "react"
import {
	createBottomTabNavigator,
	createStackNavigator
} from "react-navigation"
import Home from "./Home"
import Notifications from "./Notifications"
import ChooseDate from "./ChooseDate"
import NewLesson from "./NewLesson"
import Students from "./Students"
import NewStudent from "./NewStudent"
import Schedule from "./Schedule"
import { strings } from "../../i18n"
import { Icon } from "react-native-elements"

export default createBottomTabNavigator(
	{
		Home: Home,
		Notifications: Notifications,
		Add: {
			screen: createStackNavigator(
				{ ChooseDate: ChooseDate, NewLesson: NewLesson },
				{
					initialRouteName: "ChooseDate",
					headerMode: "none",
					navigationOptions: {
						headerVisible: false
					}
				}
			),
			navigationOptions: {
				title: "add",
				tabBarLabel: strings("tabs.add"),
				tabBarAccessibilityLabel: strings("tabs.add"),
				tabBarTestID: "NewLessonTab"
			}
		},
		Schedule: Schedule,
		Students: {
			screen: createStackNavigator(
				{ Students: Students, NewStudent: NewStudent },
				{
					mode: "modal",
					initialRouteName: "Students",
					headerMode: "none",
					navigationOptions: {
						headerVisible: false
					}
				}
			),
			navigationOptions: {
				title: "students",
				tabBarLabel: strings("tabs.students"),
				tabBarAccessibilityLabel: strings("tabs.students"),
				tabBarTestID: "StudentsTab"
			}
		}
	},
	{
		defaultNavigationOptions: ({ navigation }) => ({
			tabBarIcon: ({ focused, horizontal, tintColor }) => {
				const { routeName } = navigation.state
				let iconName
				let iconType = "material"
				if (routeName === "Home") {
					iconName = "home"
				} else if (routeName === "Notifications") {
					iconName = "notifications"
				} else if (routeName === "Add") {
					iconName = "add-circle"
				} else if (routeName === "Students") {
					iconName = "people"
				} else if (routeName == "Schedule") {
					iconName = "date-range"
				}

				// You can return any component that you like here!
				return (
					<Icon
						name={iconName}
						size={25}
						color={tintColor}
						type={iconType}
					/>
				)
			}
		}),
		tabBarOptions: {
			activeTintColor: "rgb(12,116,244)",
			inactiveTintColor: "black",
			showLabel: false
		}
	}
)
