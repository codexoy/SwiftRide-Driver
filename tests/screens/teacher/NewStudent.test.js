import "react-native"
import React from "react"

import renderer from "react-test-renderer"

import { NewStudent } from "../../../src/screens/teacher/NewStudent"

describe("Students", () => {
	test("view renders correctly", () => {
		const tree = renderer.create(<NewStudent />).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
