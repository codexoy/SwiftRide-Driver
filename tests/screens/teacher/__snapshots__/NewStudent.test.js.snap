// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Students view renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "maxHeight": 50,
        "paddingLeft": 26,
        "paddingRight": 26,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 20,
            "marginTop": 4,
          }
        }
      >
        הוספת תלמיד/ה
      </Text>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "marginLeft": "auto",
            "marginTop": -5,
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 3,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                  underlayColor="white"
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontSize": 36,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 26,
        "paddingTop": 0,
      }
    }
    testID="StudentsSearchView"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "overflow": "hidden",
          "paddingBottom": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 0,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#dcdce1",
              "borderBottomWidth": 0,
              "borderColor": "#86939e",
              "borderRadius": 30,
              "flexDirection": "row",
              "marginLeft": 0,
              "marginRight": 0,
              "minHeight": 36,
              "paddingBottom": 6,
              "paddingLeft": 8,
              "paddingTop": 6,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 8,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#7d7d7d",
                        "fontSize": 20,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <TextInput
            allowFontScaling={true}
            autoFocus={true}
            onBlur={[Function]}
            onCancel={[Function]}
            onChangeText={[Function]}
            onClear={[Function]}
            onFocus={[Function]}
            placeholder="חיפוש תלמיד"
            placeholderTextColor="#7d7d7d"
            platform="ios"
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 14,
                "marginLeft": 0,
                "minHeight": 40,
                "paddingLeft": 6,
              }
            }
            testID="searchInput"
            textAlign="right"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 15,
                "marginRight": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "right": -0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007aff",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              ביטול
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgb(212, 212, 212)",
          "borderBottomWidth": 1,
          "marginBottom": 20,
          "marginTop": 20,
          "width": "100%",
        }
      }
    />
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      keyboardShouldPersistTaps="always"
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
